server/src-rsr/initialise.sql

```diff
@@ -607,3 +596,35 @@ CREATE TABLE hdb_catalog.hdb_allowlist
   collection_name TEXT UNIQUE
     REFERENCES hdb_catalog.hdb_query_collection(collection_name)
 );
+
+CREATE TABLE hdb_catalog.remote_schema_permissions
+(
+  remote_schema TEXT,
+  role TEXT,
+  definition JSONB NOT NULL,
+  PRIMARY KEY (remote_schema, role),
+  FOREIGN KEY (remote_schema) REFERENCES hdb_catalog.remote_schemas(name) ON UPDATE CASCADE
+);
+
+CREATE VIEW hdb_catalog.hdb_permission_agg AS
```

Comment:
We can leave this definition as is and then create a view for only roles as follows:
```sql
create view hdb_roles as 
(select distinct role_name from hdb_permission),
UNION ALL
(select distinct role_name from hdb_remote_schema_permission)
```
and then you can have array relationships to `hdb_permission_agg` and `hdb_remote_schema_permission` on `hdb_roles` table.

Reply:
Cool